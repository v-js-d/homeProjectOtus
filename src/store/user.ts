import { defineStore } from "pinia";
import { computed, ref } from "vue";
import axios from "axios";
import type { loginResponse, User } from "../types/typesUser.ts";

export const useStoreUser = defineStore("user", () => {
  const user = ref(<User>{});

  const userName = computed(() => user.value?.username);

  const login = async (username: string, password: string) => {
    const res: loginResponse = await axios.post(
      `${import.meta.env.VITE_BASE_URL}/auth/login`,
      {
        username,
        password,
      },
    );
    user.value = JSON.parse(res.config.data);
    sessionStorage.setItem("token", res.data.token);
    return res
  };

  return {
    user,
    userName,
    login,
  };
});
