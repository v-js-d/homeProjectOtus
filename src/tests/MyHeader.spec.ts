import { describe, beforeEach, expect, test } from "vitest";
import component from "../components/MyHeader.vue";
import { mount } from "@vue/test-utils";
import { setActivePinia, createPinia } from "pinia";
import { useStoreUser } from "../store/user.ts";

describe("MyHeader component", () => {
  function buildWrapper(path = "/") {
    const wrapper = mount(component, {
      global: {
        stubs: ["router-link"],
        mocks: {
          $route: {
            path,
          },
        },
      },
    });
    return wrapper;
  }
  beforeEach(async () => {
    setActivePinia(createPinia());
  });

  test("mounts without errors", () => {
    const wrapper = buildWrapper("/");
    expect(wrapper.exists()).toBe(true);
    expect(wrapper.html()).toMatchSnapshot();
  });

  test("mounts when user available", async () => {
    const userStore = useStoreUser();
    userStore.user = { username: "Alex", password: "1234" };

    const wrapper = buildWrapper("/products");
    expect(wrapper.html()).toContain("Alex");
  });
});
