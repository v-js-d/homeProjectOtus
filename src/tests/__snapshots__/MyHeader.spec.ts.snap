// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MyHeader component > mounts without errors 1`] = `
"<div class="flex items-center justify-between py-3">
  <div>
    <h1>Header</h1>
  </div>
  <!--v-if-->
</div>"
`;
